processing/src/main/java/io/druid/query/topn/TopNMapFn.java

```diff
@@ -20,15 +20,62 @@
 package io.druid.query.topn;
 
 import com.google.common.base.Function;
+import io.druid.java.util.common.IAE;
 import io.druid.query.ColumnSelectorPlus;
 import io.druid.query.Result;
-import io.druid.query.topn.types.TopNStrategyFactory;
+import io.druid.query.topn.types.TopNColumnSelectorStrategyFactory;
 import io.druid.segment.Cursor;
 import io.druid.segment.DimensionHandlerUtils;
+import io.druid.segment.column.ValueType;
 
 public class TopNMapFn implements Function<Cursor, Result<TopNResultValue>>
 {
-  private static final TopNStrategyFactory STRATEGY_FACTORY = new TopNStrategyFactory();
+  public static Function<Object, Object> getValueTransformer(ValueType outputType)
+  {
+    switch (outputType) {
+      case STRING:
+        return STRING_TRANSFORMER;
+      case LONG:
+        return LONG_TRANSFORMER;
+      case FLOAT:
+        return FLOAT_TRANSFORMER;
+      default:
+        throw new IAE("invalid type: %s", outputType);
+    }
+  }
+
+  private static Function<Object, Object> STRING_TRANSFORMER = new Function<Object, Object>()
+  {
+    @Override
+    public Object apply(Object input)
+    {
+      if (input == null) {
```

Comment:
Added a convertObjectToString method in DimensionHandlerUtils and changed the 4 areas that you mentioned